### ISSUE TITLE
Better Python support

### ISSUE THREAD

**Issue by perkfly** [ErichDonGubler/vim-sublime-monokai#7]
The Python syntax highlighting is not quite the same as sublime

**Comment by ErichDonGubler** [comment]
Could you tell me how it differs? I've taken pains to make sure it's as similar as possible and to note WHY it's different where I know it -- [here](https://github.com/ErichDonGubler/vim-sublime-monokai/blob/125ed548344241ee428dab3a674c17f223912915/colors/sublimemonokai.vim#L573) are the deficiency notes as comments mixed in with the current Python highlighting. 

Another possibility is that you haven't installed [`python-mode`](https://github.com/python-mode/python-mode), which IS an undocumented dependency (my bad!). Do you already have that installed?

**Comment by perkfly** [comment]
Yeah, `python-mode`

**Comment by ErichDonGubler** [comment]
@yifeikong: I'm not sure what that means. Yes, you have it installed? Yes, installing it resolves your issue?

**Comment by godmoves** [comment]
I have the same issue, the color scheme on my computer is quite different from sublime text 3. I have installed `python-mode` but I am not sure if I made other mistakes.
Here is a comparison, the left side is sublime and the right side is vim:
![screenshot from 2018-12-02 21-14-39](https://user-images.githubusercontent.com/25997471/49340325-3f672480-f679-11e8-902c-c998b1268d4b.png)

**Comment by ErichDonGubler** [comment]
@godmoves: Thank you for the screenshot! It's very helpful. :)

So, let's address the differences I can see from your screenshot -- basically everything I see is something that either noted as a deficiency already or needs to be noted but I've already found in my testing:

* **`def` parsing is incorrect.** This is noted here:

    https://github.com/ErichDonGubler/vim-sublime-monokai/blob/125ed548344241ee428dab3a674c17f223912915/colors/sublimemonokai.vim#L580-L581
* **Function args in definition have no contextual param highlighting.** I'm...honestly not sure why this is happening to you -- it should be covered by this highlight link:

    https://github.com/ErichDonGubler/vim-sublime-monokai/blob/125ed548344241ee428dab3a674c17f223912915/colors/sublimemonokai.vim#L578-L579
* **Function calls aren't highlighted.** I'm aware of this problem, and I've unfortunately not found a syntax that totally handles this yet. I've noted this on `master`.
* **Keyword args aren't highlighted.** Yes, this is a gaping hole too, and I've noted it on `master` in https://github.com/ErichDonGubler/vim-sublime-monokai/commit/67832a6f448c12e4a1e6eecc23c2493ae5c5720b.

 I've decided to try my hand at fixing the above issues in [the `sublime` branch of my fork of `python-syntax`](https://github.com/ErichDonGubler/python-syntax/tree/sublime). You can experiment with my progress there using the `python` branch of this repo which I've just created -- I'd love to hear feedback and see what's missing!

**Comment by godmoves** [comment]
@ErichDonGubler I've fixed these issues myself and find more highlight problems. If you think it is necessary, I will create some pull  requests in your [python-syntax](https://github.com/ErichDonGubler/python-syntax/tree/sublime) repo

**Comment by ErichDonGubler** [comment]
Just so everybody is in the loop: this has fallen under my radar lately, but I'll try to get to this in the next week or so. I definitely want this theme to be Awesome™ out-of-the-box, and resolving this issue seems like a good way to do it!

**Comment by srama2512** [comment]
I wonder if there were any updates on this. I may have some additional information to add. The color scheme seems to be different across Ubuntu-18.04 and MacOS Catalina. 

This is the color scheme on Ubuntu-18.04:
![image](https://user-images.githubusercontent.com/31301440/77106856-c6b2e580-69ed-11ea-9e09-7fffbad63c10.png)

This is the color scheme on MacOS Catalina:
![image](https://user-images.githubusercontent.com/31301440/77106896-d6cac500-69ed-11ea-8660-26d84a174b78.png)

These are the relevant parts of `.vimrc` if it helps:
```
" don't bother with vi compatibility
set nocompatible

" enable column limit highlighting
set colorcolumn=80

" for easy clipboard access
set clipboard=unnamed

" enable syntax highlighting
syntax enable

set background=dark
colorscheme sublimemonokai
set termguicolors

hi Comment ctermfg=DarkRed
```

**Comment by ErichDonGubler** [comment]
@srama2512: I don't know what your second screenshot has for a color scheme, but it's not `vim-sublime-monokai`. Applying the theory it would be, there are several groups in there that I would expect to match in color (even if that color is wrong) -- for instance, `import` and `def` should both be set to Sublime's pink color, while in your second screenshot they're highlighted differently. So, I'm pretty sure the color scheme is simply not getting applied. Are you sure the `.vimrc` you've included is actually getting loaded on MacOS Catalina?

**Comment by srama2512** [comment]
Yes, you were right. The color scheme was being overridden in a different file that was causing the issue. Now things are consistent. Thanks!

**Comment by ErichDonGubler** [comment]
Alright, welp, it's been...several years, and not much progress was made by me in that time. Today, though, I took a crack at fixing up what's currently in the [`python-syntax`](https://github.com/vim-python/python-syntax) package. It got some significant gaps closed, though it didn't fix everything. I've pushed my fixes to c9552d4. It doesn't do everything, but it's _much_ better than before.

Since nobody has been forthcoming with a concrete set of cases that aren't matching expected behavior, I'm going to consider this issue "closed", and encourage folks with further divergences from Sublime to file separate issues with more details.
